File: benchmark/c/cpw/compute_n_under_modulo_p/compute_n_under_modulo_p.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int compute_n_under_modulo_p ( int n, int p ) ;
int compute_n_under_modulo_p ( int n, int p ) {
  if ( n >= p ) return 0;
  int result = 1;
  for ( int i = 1;
  i <= n;
  i ++ ) result = ( result * i ) % p;
  return result;
}

File: translation_result/else_bms/success/c/cpw/compute_n_under_modulo_p/gemini/BaseRepair/compute_n_under_modulo_p.rs

use std::num::Wrapping;

fn compute_n_under_modulo_p(n: i32, p: i32) -> i32 {
    if n >= p {
        return 0;
    }

    let mut result: Wrapping<i32> = Wrapping(1);
    for i in 1..=n {
        result = (result * Wrapping(i)) % Wrapping(p);
    }
    result.0
}
